include/mrdox/Metadata/Template.hpp

```diff
@@ -14,39 +14,254 @@
+    }
+
+    template<typename T>
+    const auto& get() const noexcept
+    {
+        using U = std::remove_cvref_t<T>;
+        if constexpr(std::is_same_v<U, TypeTParam>)
+            return Variant_.Type;
+        else if constexpr(std::is_same_v<U, NonTypeTParam>)
+            return Variant_.NonType;
+        else if constexpr(std::is_same_v<U, TemplateTParam>)
+            return Variant_.Template;
+        else
+            assert(! "invalid template parameter kind");
+    }
+};
+
+// ----------------------------------------------------------------
+
+struct TArg
+{
+    std::string Value;
+
+    TArg() = default;
+
+    MRDOX_DECL
+    TArg(std::string&& value);
+};
+
+// ----------------------------------------------------------------
+
+/** Stores information pertaining to an explicit specialization of a
+    member of an implicitly instantiated class template specialization.
+
+    This structure is stored in the `TemplateInfo` corresponding to the
+    outermost specialized template. If the explicitly specialized
+    member is itself a member of a nested template, the `SpecializationInfo`
+    which stores the arguments for the parent template will additionally
+    store a pointer to the SpecializationInfo for the nested template,
+    recursively. Each `SpecializationInfo` node contains a vector of
+    `SymbolID` pairs `(specialized, primary)`, where `specialized` is the
+    replacement definition of `primary` for the given set of template
+    arguments of its parent template(s).
+*/
+struct SpecializationInfo
+{
+    /** The template arguments the parent template is specialized for */
+    std::vector<TArg> Args;
+
+    /** ID of the template to which the arguments pertain */
+    SymbolID Template;
+
+    /** SpecializationInfo for nested templates which are also specialized */
+    std::vector<SpecializationInfo> Nested;
+
+    /** SymbolID pairs of any specialized members.
+        the first element is the ID of the original member (i.e. member that is replaced)
+        the second element is the ID of the specialized member (i.e. the replacement)
+    */
+    std::vector<std::pair<SymbolID, SymbolID>> Members;
+};
+
+// ----------------------------------------------------------------
+
 enum class TemplateSpecKind
 {
     Primary = 0, // for bitstream
     Explicit,
     Partial
 };
 
-// stores information pertaining to primary template and
-// partial/explicit specialization declarations
+/** Stores information pertaining to primary template and
+    partial/explicit specialization declarations
+*/
 struct TemplateInfo
 {
-    // for primary templates:
-    //     - Params will be non-empty
-    //     - Args will be empty
-    // for explicit specializations:
-    //     - Params will be empty
-    // for partial specializations:
-    //     - Params will be non-empty
-    //     - each template parameter will appear at least
-    //       once in Args outside of a non-deduced context
+    /** For primary templates:
```

Comment:
The brief is different from the description. Add a proper brief and reformat the description to be a list (use `@li` for list items)

Reply:
I'm going to change this to be a non-javadoc comment as it does not apply to a single data member.